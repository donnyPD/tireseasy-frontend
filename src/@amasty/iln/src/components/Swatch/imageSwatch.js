import React, { useMemo } from 'react';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import { bool, shape, string } from 'prop-types';

import defaultClasses from './swatch.css';
import Icon from '@magento/venia-ui/lib/components/Icon';
import { Check as CheckIcon } from 'react-feather';
import { generateUrl } from '@magento/peregrine/lib/util/imageUtils';

const SWATCH_WIDTH = 96;

const ImageSwatch = props => {
  const { isSelected, style, swatchData, label } = props;
  const classes = mergeClasses(defaultClasses, props.classes);

  const icon = useMemo(() => {
    return isSelected ? <Icon src={CheckIcon} /> : null;
  }, [isSelected]);

  const finalStyle = useMemo(() => {
    let result = style;

    if (swatchData) {
      let swatchValue = swatchData;

      if (!swatchData.startsWith('#')) {
        const imagePath = generateUrl(swatchData, 'image-swatch')(SWATCH_WIDTH);

        swatchValue = `url("${imagePath}")`;
      }

      result = Object.assign({}, style, {
        '--venia-swatch-bg': swatchValue
      });
    }

    return result;
  }, [style, swatchData]);

  const className = classes[`imageSwatch${isSelected ? '_selected' : ''}`];

  return (
    <span className={className} style={finalStyle}>
      {icon}
      <span className={classes.tooltip}>
        <span className={classes.image} style={finalStyle} />
        <span className={classes.label}>{label}</span>
      </span>
    </span>
  );
};

ImageSwatch.propTypes = {
  isSelected: bool,
  swatchData: string,
  label: string,
  classes: shape({
    root: string
  })
};

export default ImageSwatch;
