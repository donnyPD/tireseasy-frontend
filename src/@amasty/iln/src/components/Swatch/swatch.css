.root {
  padding: 0;
  margin: 0;
  min-width: 3rem;
  height: 3rem;
  position: relative;
}

.textSwatch {
  composes: root from '~@magento/venia-ui/lib/components/ProductOptions/tile.module.css';
}

.textSwatch_selected {
  composes: root_selected from '~@magento/venia-ui/lib/components/ProductOptions/tile.module.css';
}

.imageSwatch {
  composes: root from '~@magento/venia-ui/lib/components/ProductOptions/swatch.module.css';
  background-size: contain;
  background-position: center center;
  background-repeat: no-repeat;
  height: 3rem;
}

.imageSwatch_selected {
  composes: imageSwatch;
  position: relative;
}

.imageSwatch_selected:after {
  content: '';
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: linear-gradient(-45deg, rgba(0, 0, 0, 0.2), transparent);
}

.tooltip {
  composes: bg-white from global;
  display: none;
  position: absolute;
  bottom: 110%;
  left: 50%;
  transform: translateX(-50%);
  width: auto;
  min-width: 4rem;
  max-width: 10rem;
  text-align: center;
  z-index: 1;
  padding: 0.5rem;
  color: rgb(var(--venia-global-color-text));
  border-radius: 1px;
  box-shadow: 0 1px 6px rgb(var(--venia-global-color-gray-900) / 0.2);
}

.root:hover .tooltip {
  display: block;
}

.image {
  display: block;
  width: 100%;
  min-width: 6rem;
  height: 6rem;
  background: var(--venia-swatch-bg);
  margin-bottom: 0.5rem;
  background-size: contain;
  background-repeat: no-repeat;
}

.label {
  display: block;
  font-size: 0.875rem;
  line-height: 1rem;
}
