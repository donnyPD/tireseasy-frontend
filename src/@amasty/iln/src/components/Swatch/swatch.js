import React from 'react';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import { shape, string } from 'prop-types';
import defaultClasses from './swatch.css';
import ImageSwatch from './imageSwatch';
import TextSwatch from './textSwatch';

const Swatch = props => {
  const {
    isSelected,
    item: { label, image },
    onClick,
    textOnly
  } = props;

  const classes = mergeClasses(defaultClasses, props.classes);
  const className = classes[`root${isSelected ? '_selected' : ''}`];

  const child =
    !textOnly && image ? (
      <ImageSwatch isSelected={isSelected} label={label} swatchData={image} />
    ) : (
      <TextSwatch label={label} isSelected={isSelected} swatchData={image} />
    );

  return (
    <button className={className} title={label} onClick={onClick} type="button">
      {child}
    </button>
  );
};

Swatch.propTypes = {
  classes: shape({
    root: string
  })
};

export default Swatch;
