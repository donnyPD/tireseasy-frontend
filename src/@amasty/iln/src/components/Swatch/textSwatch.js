import React from 'react';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import { shape, string, bool } from 'prop-types';

import defaultClasses from './swatch.css';

const TextSwatch = props => {
  const { label, isSelected, swatchData } = props;
  const classes = mergeClasses(defaultClasses, props.classes);

  const className = classes[`textSwatch${isSelected ? '_selected' : ''}`];

  return (
    <span className={className}>
      {swatchData || label}
      <span className={classes.tooltip}>
        <span className={classes.label}>{label}</span>
      </span>
    </span>
  );
};

TextSwatch.propTypes = {
  label: string,
  isSelected: bool,
  swatchData: string,
  classes: shape({
    root: string
  })
};

export default TextSwatch;
