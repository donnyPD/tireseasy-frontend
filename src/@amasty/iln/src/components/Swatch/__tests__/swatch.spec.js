import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import Swatch from '../swatch';

jest.mock('../imageSwatch', () => 'ImageSwatch');
jest.mock('../textSwatch', () => 'TextSwatch');

test('render text swatch', () => {
  const props = {
    item: {
      label: 'test',
      count: 4
    }
  };
  const tree = createTestInstance(<Swatch {...props} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('render image swatch', () => {
  const props = {
    item: {
      label: 'test',
      count: 4,
      image: 'imgSrc'
    }
  };
  const tree = createTestInstance(<Swatch {...props} />);
  expect(tree.toJSON()).toMatchSnapshot();
});
