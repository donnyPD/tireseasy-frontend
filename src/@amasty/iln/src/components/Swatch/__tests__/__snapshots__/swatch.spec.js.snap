// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render image swatch 1`] = `
<button
  title="test"
  type="button"
>
  <ImageSwatch
    label="test"
    swatchData="imgSrc"
  />
</button>
`;

exports[`render text swatch 1`] = `
<button
  title="test"
  type="button"
>
  <TextSwatch
    label="test"
  />
</button>
`;
